### ISSUE TITLE
ApplicationIcon PropertyGroup doesn't seem to work for F#.  Net Core 3.0

### ISSUE THREAD

**Issue by nickybaby** [dotnet/fsharp#10065]
# ApplicationIcon PropertyGroup doesn't seem to work as intended for F#.  Net Core 3.0

# General
.Net Core 3 / Windows 10

I have this in my .fsproj:
```
<PropertyGroup>
  <ApplicationIcon>favicon.ico</ApplicationIcon>
</PropertyGroup>
```

This compiles but doesn't change the icon as intended.  The favicon is in the same directory structure as my .fsproj file, but I have tried a fully qualified path instead of relative path structure.

This is my first filed issue, so I apologize if I misunderstood the contribution guidelines or if this is the wrong place.  Please just let me know and I will modify, or redirect, immediately.

**Comment by baronfel** [comment]
Some research/context that might be useful for anyone interested:

* ApplicationIcon is passed as one of a set of Win32 resources to the C#/VB compilers in the targets [here](https://github.com/dotnet/roslyn/blob/afd10305a37c0ffb2cfb2c2d8446154c68cfa87a/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets#L146-L148)
* These target properties are forwarded to the underlying compiler as arguments [here](https://github.com/dotnet/roslyn/blob/87ce1bf93fbc52ff09dbf2c6cd8b0d1603b6f652/src/Compilers/Core/MSBuildTask/Csc.cs#L576-L597)
* ApplicationIcon and Win32Manifest as a pair are mutually exclusive with a user-provided Win32Resource, the logic of which is [here](https://github.com/dotnet/roslyn/blob/425c7397403f4b9587e54f583fe8bdaf30b73468/src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs#L1508-L1535)
* Actually creating the resource streams for the icon/app manifest happens [here](https://github.com/dotnet/roslyn/blob/425c7397403f4b9587e54f583fe8bdaf30b73468/src/Compilers/Core/Portable/Compilation/Compilation.cs#L1590-L1634)

So basically this logic would need to be ported to the F# targets, FSC task, and internal compiler resource generation.

**Comment by albert-du** [comment]
Interestingly enough the Roslyn logic for resource generation was already added https://github.com/dotnet/fsharp/blob/main/src/fsharp/absil/ilnativeres.fs#L631.
